### PR TITLE
Onboarding exercises

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
Do not merge

### CODE REVIEW

File: sync/sync.go [hunk 1]
```diff
@@ -97,10 +107,114 @@ func (s *Synchronizer) WithListener(listener EventListener) *Synchronizer {
...
 
+func (s *Synchronizer) syncBlocks2(syncCtx context.Context) {
+	streamCtx, streamCancel := context.WithCancel(syncCtx)
+	i := s.nextHeight()
+	for ; ; i++ {
+		select {
+		case <-streamCtx.Done():
+			streamCancel()
+			return
+		default:
```
Comment (kirugan) [inline sync/sync.go:122]: If there is a failure in one of the statements below: we should immediately stop syncing (it's up to you how to resolve it `continue` or something else)

Comment (AnkushinDaniil) [reply sync/sync.go:122]: [fixed](https://github.com/NethermindEth/juno/pull/1934/commits/f1846fce756bde5aa6e91f3062e38c9095865465)

**Comment by rianhughes** [comment]
Looks good, nice work!

**Comment by kirugan** [comment]
all good - closing the PR

**Pull request closed without merging**